ui/pages/confirmations/components/confirm/info/approve/hooks/use-approve-token-simulation.ts

```diff
@@ -38,23 +37,24 @@ export const useApproveTokenSimulation = (
         !isBoolean(param.value),
     );
     if (paramIndex === -1) {
-      return 0;
+      return '0';
     }
 
-    return new BigNumber(value.data[0].params[paramIndex].value.toString())
-      .dividedBy(new BigNumber(10).pow(Number(decimals)))
-      .toNumber();
+    return calcTokenAmount(
+      value.data[0].params[paramIndex].value,
+      Number(decimals),
+    ).toFixed();
   }, [value, decimals]);
 
   const formattedSpendingCap = useMemo(() => {
     // formatting coerces small numbers to 0
-    return isNFT || decodedSpendingCap < 1
-      ? toNonScientificString(decodedSpendingCap)
-      : new Intl.NumberFormat(locale).format(decodedSpendingCap);
+    return isNFT || parseInt(decodedSpendingCap, 10) < 1
+      ? decodedSpendingCap
+      : new Intl.NumberFormat(locale).format(parseInt(decodedSpendingCap, 10));
```

Comment:
This might cause some precision issues with parseInt here. Examples of supporting precision passing BN to Intl.NumberFormat can be found here
ui/pages/confirmations/components/simulation-details/formatAmount.ts